lang/cs/Org.Apache.REEF.Network.Tests/GroupCommunication/GroupCommunicationTests.cs

```diff
@@ -380,6 +380,45 @@ public void TestBroadcastOperator()
             Assert.Equal(value, receiver2.Receive());
         }
 
+        /// <summary>
+        /// Test IBroadcastReceiver.Receive() with and without cancellation token
+        /// </summary>
+        [Fact]
+        public void TestBroadcastOperatorWithCancelation()
+        {
+            string groupName = "group1";
+            string operatorName = "broadcast";
+            string masterTaskId = "task0";
+            string driverId = "Driver Id";
+            int numTasks = 10;
+            int value = 1337;
+            int fanOut = 3;
+
+            IGroupCommDriver groupCommDriver = GetInstanceOfGroupCommDriver(driverId, masterTaskId, groupName, fanOut, numTasks);
+
+            var commGroup = groupCommDriver.DefaultGroup
+                .AddBroadcast(operatorName, masterTaskId)
+                .Build();
+
+            List<ICommunicationGroupClient> commGroups = CommGroupClients(groupName, numTasks, groupCommDriver, commGroup, GetDefaultCodecConfig());
+
+            IBroadcastSender<int> sender = commGroups[0].GetBroadcastSender<int>(operatorName);
+            IBroadcastReceiver<int> receiver1 = commGroups[1].GetBroadcastReceiver<int>(operatorName);
+            IBroadcastReceiver<int> receiver2 = commGroups[2].GetBroadcastReceiver<int>(operatorName);
+
+            Assert.NotNull(sender);
+            Assert.NotNull(receiver1);
+            Assert.NotNull(receiver2);
+
+            sender.Send(value);
+            Assert.Equal(value, receiver1.Receive());
+
+            var token = new CancellationTokenSource();
```

Comment:
This just checks that cancellation happened once the token was already cancelled.....I would have another case: make sender not send anything, run receivers in separate threads so that they are blocked on queue and then cancel the token and check for exception/...does that make sense?